.github/actions/docker-push-variables/action.yml

```diff
@@ -51,12 +51,14 @@ runs:
         version_family = ".".join(version.split(".")[:-1])
         safe_branch_name = branch_name.replace("refs/heads/", "").replace("/", "-")
 
+        sha = os.environ["GITHUB_SHA"] if not "${{ github.event.pull_request.head.sha }}" else "${{ github.event.pull_request.head.sha }}"
```

Comment:
I suppose there isn't a way to do this purely with environment variables

Reply:
unfortunately no :/